src/persistentstorage.cpp

```diff
@@ -258,3 +266,229 @@ void MysqlPersistentStorage::writelog(const std::string &id, const ObjectValue &
     }
 }
 #endif // HAVE_MYSQL
+
+
+#ifdef HAVE_INFLUXDB
+Logger& InfluxdbPersistentStorage::logger_m(Logger::getInstance("InfluxdbPersistentStorage"));
+
+bool _existsInArray(const Json::Value &json, const std::string &key)
+{
+    bool ret = false;
+    assert(json.isArray());
+    for (int i = 0; i != json.size(); i++)
+    {
+        if (json[i][0].asString() == key)
+            ret = true;
+    }
+    return ret;
+}
+
+bool InfluxdbPersistentStorage::createDB(const std::string &db, std::string &result)
+{
+    std::string query = "q=CREATE DATABASE ";
+    query.append(db);
+    return curlRequest(INFLUXDB_QUERY, db, query, result);
+}
+
+InfluxdbPersistentStorage::InfluxdbPersistentStorage(ticpp::Element* pConfig)
+{
+    uri_m = pConfig->GetAttribute("uri");
+    user_m = pConfig->GetAttribute("user");
+    pass_m = pConfig->GetAttribute("pass");
+    db_m = pConfig->GetAttribute("db");
+    persist_db_m = pConfig->GetAttribute("persist_db");
```

Comment:
@fraxinas `persistenceDb_m` would be more in line with the naming convention in this codebase. No snake case.